server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1743,7 +1743,7 @@ public final boolean shouldProcessCheckpoint(ReplicationCheckpoint requestCheckp
             return false;
         }
         ReplicationCheckpoint localCheckpoint = getLatestReplicationCheckpoint();
-        if (localCheckpoint.isAheadOf(requestCheckpoint)) {
+        if (requestCheckpoint.isAheadOf(getLatestReplicationCheckpoint()) == false) {
```

Comment:
Refactored this logic because the equality check was giving false negatives by comparing all fields inside ReplicationCheckpoint when all we want is pterm and SI version, so on processLatestcheckpoint it would redundantly process the same checkpoint over and over.

Reply:
Can you use the `localCheckpoint` variable here instead of calling `getLatestReplicationCheckpoint()` again?